common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -335,6 +347,23 @@ protected void processRedirectUrl(@NonNull final WebView view, @NonNull final St
         //the TokenTask should be processed at after the authorization process in the upper calling layer.
     }
 
+    /**
+     * Launch the browser with the given action URI and code.
+     * <p>
+     * From the query parameters, extract the action URI and code,
+     * The constructs the URI with the action URI and code.
+     *
+     * @param url The URL to be opened in the browser.
+     */
+    private boolean processSwitchBrowserRequest(@NonNull final String url) {
+        final SwitchBrowserChallenge switchBrowserChallenge =
+                SwitchBrowserChallenge.constructFromRedirectUri(Uri.parse(url));
+        if (switchBrowserChallenge == null) {
+            return false;
```

Comment:
What is the end user impact of this?

Reply:
reworked to throw exception in case of malformed URL